libraries/AP_Scripting/generator/src/main.c

```diff
@@ -1310,6 +1324,57 @@ void emit_userdata_fields() {
   }
 }
 
+// emit refences functions for a call, return the number of arduments added
+int emit_references(const struct argument *arg, const char * tab) {
+  int arg_index = NULLABLE_ARG_COUNT_BASE + 2;
+  int return_count = 0;
+  while (arg != NULL) {
+    if (arg->type.flags & (TYPE_FLAGS_NULLABLE | TYPE_FLAGS_REFERNCE)) {
+      return_count++;
+      switch (arg->type.type) {
+        case TYPE_BOOLEAN:
+          fprintf(source, "%slua_pushboolean(L, data_%d);\n", tab, arg_index);
+          break;
+        case TYPE_FLOAT:
+          fprintf(source, "%slua_pushnumber(L, data_%d);\n", tab, arg_index);
+          break;
+        case TYPE_INT8_T:
+        case TYPE_INT16_T:
+        case TYPE_INT32_T:
+        case TYPE_UINT8_T:
+        case TYPE_UINT16_T:
+        case TYPE_ENUM:
+          fprintf(source, "%slua_pushinteger(L, data_%d);\n", tab, arg_index);
+          break;
+        case TYPE_UINT32_T:
+          fprintf(source, "%snew_uint32_t(L);\n", tab);
+          fprintf(source, "%s*static_cast<uint32_t *>(luaL_checkudata(L, -1, \"uint32_t\")) = data_%d;\n", tab, arg_index);
+          break;
+        case TYPE_STRING:
+          fprintf(source, "%slua_pushstring(L, data_%d);\n", tab, arg_index);
+          break;
+        case TYPE_USERDATA:
+          // userdatas must allocate a new container to return
+          fprintf(source, "%snew_%s(L);\n", tab, arg->type.data.ud.sanatized_name);
+          fprintf(source, "%s*check_%s(L, -1) = data_%d;\n", tab, arg->type.data.ud.sanatized_name, arg_index);
+          break;
+        case TYPE_NONE:
+          error(ERROR_INTERNAL, "Attempted to emit a nullable argument of type none");
```

Comment:
```suggestion
          error(ERROR_INTERNAL, "Attempted to emit a nullable or reference argument of type none");
```
Either make the error here more clear, or pass the string in, in this case I think that the change I suggested is fine.

Reply:
done